packages/frontend/app/components/reports/new-subject.js

```diff
@@ -269,18 +269,33 @@ export default class ReportsNewSubjectComponent extends Component {
     return this.usersPrimarySchool;
   }
 
+  get includeAnythingObject() {
+    const exceptedSubjects = ['instructor', 'mesh term'];
+    return !exceptedSubjects.includes(this.subject);
+  }
+
   @action
   changeSubject(subject) {
     this.selectedSubject = subject;
-    this.selectedPrepositionalObject = null;
-    this.selectedPrepositionalObjectId = null;
-    this.clearErrorDisplay();
+
+    if (this.includeAnythingObject) {
+      this.changePrepositionalObject(null);
+    } else {
+      const firstPrepositionalObjectLabel = Object.values(this.prepositionalObjectList)
+        .map((item) => item.label)
+        .sort()[0];
+      const firstPrepositionalObject = this.fullPrepositionalObjectList.filter(
+        (item) => item.label == firstPrepositionalObjectLabel,
```

Comment:
please use triple equal here for strict comparison, a bit surprised that the linter doesn't complain here.

```suggestion
        (item) => item.label === firstPrepositionalObjectLabel,
```